### PR TITLE
dev/drupal#22: implement support for Drupal8 views and multilingual C…

### PR DESCRIPTION
**Pull request by mlutfy**
This is a (mostly copy-pasted) port of the Drupal7 views code.

### CODE REVIEW

File: civicrm.views_execution.inc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function civicrm_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  \Drupal::service('civicrm')->initialize();
+
+  // check if we are in multilingual mode, otherwise return
+  // TODO: should be a simple call - is_multiligual ?
+  $domain = new CRM_Core_DAO_Domain();
```
Comment (seamuslee001) [inline civicrm.views_execution.inc:14]: @mlutfy does this need a \ at the start to say we should be in the root namespace?

Comment (mlutfy) [reply civicrm.views_execution.inc:14]: I don't think it's required/suggested? It seems to work in both cases, but grepping around, I don't see `new \CRM`. The only place I see it, is under `civicrm-core/Civi/`.

Correct me if I'm wrong, but googling around, people seem to suggest avoiding it if not required, for code readability. It should only be used when there is a risk of a namespace clash between the local class and global scope.

(sorry for the slow response, had not seen the email notification on this thread)

**Comment by jackrabbithanna** [comment]
Just wanted to let you'll know that the latest CiviCRM Entity 8.x-3.x-dev version includes Views support....Not telling you what to do, but the "Core Views Integration" could be handled by that module....

**Comment by mlutfy** [comment]
I opened https://lab.civicrm.org/dev/drupal/issues/28 to discuss Views deprecation from Core. Seems like a good idea.

Nonetheless, until that happens, can we merge this? I'll also start experimenting with CiviCRM Entity and test how it handle multi-lingual.

**Comment by jackrabbithanna** [comment]
I'd say merge it if it works...It is likely CiviCRM Entity will need some portion of Views handler code ported over from the core Views integration module to reach feature parity....

It will likely be a lot of copy and pasting, not everything needs to be rewritten, but HOW it is integrated may be changed a little bit...

We'll see what the feedback is on #28 before a large amount of time is spent on this, but now is the time to discuss a switch over as a community, before any official releases go out...

**Comment by JoeMurray** [comment]
mlufty could you try creating views with civi d8 entity and entity views integration? It worked for us for simple cases and it would be good to consolidate development efforts on one approach. Leveraging more D8 code and supporting less civi specific code seems wiser given our linited resources.

File: civicrm.views_execution.inc (hunk 1 above)
Comment (borisson) [reply civicrm.views_execution.inc:14]: It's not required, but it is more explicit and that's a great thing. It's also marginally faster.

**Comment by borisson** [comment]
It seems like the conclusion there is to remove it from this repo. So I think this PR should be closed and a new one opened to remove all the views integration from this module?

**Comment by mlutfy** [comment]
It's not on the top of my list, but I'll close the PR for now. We can continue discussion here: https://lab.civicrm.org/dev/drupal/issues/22

**Comment by mlutfy** [comment]
For anyone stumbling on this issue: CiviCRM Entity does handle multilingual views without any issues.

**Pull request closed without merging**